### PR TITLE
Develop/8

### PR DESCRIPTION
**Pull request by devmin222**

### CODE REVIEW

File: resources/views/pages/admin/banner/banner-dashboard.blade.php [hunk 1]
```diff
@@ -101,7 +101,7 @@
                     <td>
                         @if($banner->status == 0)노출
 
-                        @else 미노출
+                        @else
```
Comment (devmin222) [inline resources/views/pages/admin/banner/banner-dashboard.blade.php:104]: Why has else been removed? Also can use ternary here?

File: resources/views/pages/admin/resource/resource-dashboard.blade.php [hunk 2]
```diff
@@ -93,70 +93,35 @@
...
-                        </div>
-                    </td>
-                    <td>700</td>
-                    <td>미노출</td>
-                    <td>2023.01.23</td>
-                </tr>
-                <tr>
-                    <td>7</td>
-                    <td><a href="{{ url('admin/resourceDetails') }}" class="td-a-custom">가지에 만물은 수 앞이 맺어, 들어 그리하였는가? 우는 인생을 굳세게 황금시대다. 피가 인생에 그들은 말이다.</a></td>
+
```
Comment (devmin222) [inline resources/views/pages/admin/resource/resource-dashboard.blade.php:96]: Clean up the code formatting here

File: resources/views/pages/admin/resource/resource-details.blade.php [hunk 3]
```diff
@@ -45,33 +45,29 @@
                         <div class="height-52 item-flex-align-start ps-4">
                             <div>
                                 <div class="item-flex-start mb-1">
+                                    <ul>
+                                    @foreach(json_decode($resource->path) as $filePath)
```
Comment (devmin222) [inline resources/views/pages/admin/resource/resource-details.blade.php:49]: Clean up code formatting

File: resources/views/pages/admin/resource/resource-modification.blade.php [hunk 4]
```diff
@@ -51,38 +55,32 @@
...
-                                <span class="position-rel"><i class='far fa-file-alt view-file-i'></i></span>
-                                <input type="text" class="form-control val-text file-up-bar-custom" name="file1" id="file1" placeholder="" value="자료123.pdf" aria-describedby="File Up">
-                            </div>
-                            <a href="#deleteConfirmationModal" class="btn btn12 ms-4" data-bs-toggle="modal">파일 삭제</a>
-                        </div>
-                        <div class="height-52 item-flex-start ml30 my-3">
-                            <div class="height-52">
-                                <span class="position-rel"><i class='far fa-file-alt view-file-i'></i></span>
-                                <input type="text" class="form-control val-text file-up-bar-custom" name="file1" id="file1" placeholder="" value="자료123456.pdf" aria-describedby="File Up">
+
```
Comment (devmin222) [inline resources/views/pages/admin/resource/resource-modification.blade.php:70]: Clean up code

File: resources/views/pages/resources.blade.php [hunk 5]
```diff
@@ -88,126 +88,32 @@
...
-                                <td style="width: 55%; text-align: left;">
-                                    <a href="resources_details.php">가지에 만물은 수 앞이 맺어, 들어 그리하였는가? 우는 인생을 굳세게 황금시대다. 피가 인생에 그들은 말이다.
-                                    </a>
-                                </td>
-                                <td style="width: 5%">
-                                    <img src="images/pdf.png" style="width: 18px; height: 18px;">
-                                </td>
-                                <td style="width: 15%">2023.02.01</td>
-                                <td style="width: 15%">300</td>
+                                    <ul>
```
Comment (devmin222) [inline resources/views/pages/resources.blade.php:101]: Clean up code

Comment (Dilshanmuditha) [reply resources/views/pages/resources.blade.php:100]: done

File: resources/views/pages/admin/resource/resource-modification.blade.php (hunk 4 above)
Comment (Dilshanmuditha) [reply resources/views/pages/admin/resource/resource-modification.blade.php]: done

File: resources/views/pages/admin/resource/resource-details.blade.php (hunk 3 above)
Comment (Dilshanmuditha) [reply resources/views/pages/admin/resource/resource-details.blade.php]: done

File: resources/views/pages/admin/resource/resource-dashboard.blade.php (hunk 2 above)
Comment (Dilshanmuditha) [reply resources/views/pages/admin/resource/resource-dashboard.blade.php]: done

File: resources/views/pages/admin/banner/banner-dashboard.blade.php (hunk 1 above)
Comment (Dilshanmuditha) [reply resources/views/pages/admin/banner/banner-dashboard.blade.php]: done

File: app/Http/Controllers/ResourceController.php [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            } */
+        Storage::deleteDirectory('public/files/resources/'.$id);
+        $resource->delete();
+
+        return redirect('/admin/resourceDash')->with('resource delete','Resource deleted successfully');
+    }
+
+    public function resourceUpdate(Request $request, $id )
+    {
+
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/ResourceController.php:71]: done

File: app/Http/Controllers/ResourceController.php [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        $validator = Validator::make($request->all(), [
+            'title' => 'required',
+            'contents' => 'required',
+            'files' => 'required',
+            'files.*' => 'mimes:jpeg,png,pdf|max:2048',
+        ]);
+        
+
+        if ($request->hasFile('files')) {
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/ResourceController.php]: done

File: app/Http/Controllers/ResourceController.php [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+{
+    public function resourceAdd(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'title' => 'required',
+            'contents' => 'required',
+            'files' => 'required',
+            'files.*' => 'mimes:jpeg,png,pdf|max:2048',
+        ]);
+
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/ResourceController.php:21]: done

File: app/Http/Controllers/BannerController.php
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/BannerController.php:1]: done

File: app/Http/Controllers/AdminPageController.php [hunk 9]
```diff
@@ -183,23 +184,32 @@ public function directionModify()
...
     public function resourceDashboard()
     {
-        return view('pages.admin.resource.resource-dashboard');
+        $resources = Resource::orderBy('id','desc')->get();
+        return view('pages.admin.resource.resource-dashboard',compact('resources'));
     }
     public function resourceRegistration()
     {
-        return view('pages.admin.resource.resource-register');
+        $exposure = Resource::pluck('status')->toArray();
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/AdminPageController.php]: done it.

File: app/Http/Controllers/ResourceController.php [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+                if ($file->isValid()) {
+                    $name = $file->getClientOriginalName();
+                    $path = $file->storeAs('public/files/resources/'.$id,$name);
+                    $filePaths[] = $path;
+                }
+            }
+            
+            $data->path = json_encode($filePaths);
+        }
+            $data->status = $request->input('exposureStatus');
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply app/Http/Controllers/ResourceController.php]: done

File: database/migrations/2023_06_05_044821_create_resources_table.php [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('resources', function (Blueprint $table) {
+            $table->id();
+            $table->integer('status');
+            $table->string('title');
+            $table->string('path');
+            $table->string('views')->nullable();
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply database/migrations/2023_06_05_044821_create_resources_table.php]: done

File: database/migrations/2023_06_05_044821_create_resources_table.php [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('resources', function (Blueprint $table) {
+            $table->id();
+            $table->integer('status');
+            $table->string('title');
+            $table->string('path');
```
[the remark this answers is not in the document]
Comment (Dilshanmuditha) [reply database/migrations/2023_06_05_044821_create_resources_table.php]: done

**Review by devmin222 — COMMENTED** [review]
Good, but need to improve

**Pull request merged**